components/oauth-extensions/dynamic-client-manager/src/main/java/org/wso2/carbon/identity/oauth/extension/impl/ClientRegistrationServiceImpl.java

```diff
@@ -232,4 +250,45 @@ public OAuthApplicationInfo createOAuthApplication(
         return null;
     }
 
+    public void unregisterApplication(String userId, String applicationName, String consumerKey)
+            throws APIManagementException {
+
+        String tenantDomain = MultitenantUtils.getTenantDomain(userId);
+        String baseUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        String userName = MultitenantUtils.getTenantAwareUsername(userId);
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
+
+        if (userId == null || userId.isEmpty()) {
+            throw new APIManagementException("Error occurred while unregistering Application: userId cannot be null/empty");
+        }
+        try {
+            OAuthAdminService oAuthAdminService = new OAuthAdminService();
+            OAuthConsumerAppDTO oAuthConsumerAppDTO = oAuthAdminService.getOAuthApplicationData(consumerKey);
```

Comment:
It is used to check whether the application is already there for given consumer key.